File: kmp/composeApp/src/commonMain/kotlin/dev/jianastrero/trainer/domain/model/pokeapi/response/PokemonItemResponse.kt
package dev.jianastrero.trainer.domain.model.pokeapi.response

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PokemonItemResponse(
    @SerialName("name") val name: String,
    @SerialName("url") val url: String
) {
    val id: String
        get() = url.split("/").last { it.isNotEmpty() }

    val imageUrl: String
        get() = "$POKEMON_OFFICIAL_ARTWORK_BASE_URL/$id.$POKEMON_OFFICIAL_ARTWORK_EXTENSION"

    companion object {
        private const val POKEMON_OFFICIAL_ARTWORK_BASE_URL =
            "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork"
        private const val POKEMON_OFFICIAL_ARTWORK_EXTENSION = "png"
    }
}

File: react_native/src/domain/model/pokeapi/response/PokemonItemResponse.ts
export default class PokemonItemResponse {
    public name: string
    public url: string

    constructor(name: string, url: string) {
        this.name = name;
        this.url = url;
    }

    getId(): string {
        const splits = this.url.split('/');
        return splits[splits.length - 2];
    }

    getImageUrl(): string {
        return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${this.getId()}.png`;
    }
}
